presto-parquet/src/main/java/com/facebook/presto/parquet/cache/MetadataReader.java

```diff
@@ -112,9 +126,52 @@ public static ParquetFileMetadata readFooter(ParquetDataSource parquetDataSource
             tailSlice = wrappedBuffer(footerBuffer, 0, footerBuffer.length);
         }
 
-        FileMetaData fileMetaData = readFileMetaData(tailSlice.slice(tailSlice.length() - completeFooterSize, metadataLength).getInput());
+        return readParquetMetadata(tailSlice.slice(tailSlice.length() - completeFooterSize, metadataLength).getInput(), metadataLength, modificationTime, fileDecryptor, encryptedFooterMode, parquetDataSource.getId());
+    }
+
+    private static ParquetFileMetadata readParquetMetadata(BasicSliceInput input, int metadataLength, long modificationTime, Optional<InternalFileDecryptor> fileDecryptor, boolean encryptedFooterMode, ParquetDataSourceId id)
+            throws IOException
+    {
+        if (encryptedFooterMode && !fileDecryptor.isPresent()) {
+            new IllegalArgumentException("fileDecryptionProperties cannot be null when encryptedFooterMode is true");
+        }
+        Decryptor footerDecryptor = null;
+        byte[] aad = null;
```

Comment:
how about we add comment for aad:
`additional authenticated data for AES cipher`